nexus/src/app/background/plan_execution.rs

```diff
@@ -0,0 +1,327 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Background task for realizing a plan blueprint
+
+use super::common::BackgroundTask;
+use anyhow::Context;
+use futures::future::BoxFuture;
+use futures::stream;
+use futures::FutureExt;
+use futures::StreamExt;
+use nexus_db_queries::context::OpContext;
+use nexus_types::blueprint::{Blueprint, OmicronZonesConfig};
+use serde_json::json;
+use slog::Logger;
+use std::collections::BTreeMap;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::sync::watch;
+use uuid::Uuid;
+
+/// Background task that takes a [`Blueprint`] and realizes the change to
+/// the state of the system based on the `Blueprint`.
+pub struct PlanExecutor {
+    rx_blueprint: watch::Receiver<Option<Arc<Blueprint>>>,
+}
+
+impl PlanExecutor {
+    // Temporary until we wire up the background task
+    #[allow(unused)]
+    pub fn new(
+        rx_blueprint: watch::Receiver<Option<Arc<Blueprint>>>,
+    ) -> PlanExecutor {
+        PlanExecutor { rx_blueprint }
+    }
+}
+
+impl BackgroundTask for PlanExecutor {
+    fn activate<'a, 'b, 'c>(
+        &'a mut self,
+        opctx: &'b OpContext,
+    ) -> BoxFuture<'c, serde_json::Value>
```

Comment:
Hmm, this can just be `activate<'a>(&'a mut self, opctx: &'a OpContext) -> BoxFuture<'a, serde_json::Value>`. That's because all the individual lifetime parameters here are [covariant](https://lifetime-variance.sunshowers.io/ch01-02-formalizing-variance.html).

Reply:
https://github.com/oxidecomputer/omicron/pull/4877